### PR TITLE
✨ Support redelegation function

### PR DESCRIPTION
**Pull request by WeiJun0827**

### CODE REVIEW

**Comment by williamchong** [comment]
image
image
@nwingt Do you think this is ok or do you want to further improve the UI?

**Comment by williamchong** [comment]
image
Strange that there are two active validators mixed into inactive validators

File: components/Modal/Restake.vue [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      field-id="to"
+      field-label="To"
+    >
+      <select id="to" v-model="destinationValidator">
+        <option disabled value="">Destination Validator</option>
+        <option
+          v-for="(validator, index) in sortedEnrichedValidators"
+          :key="validator.operatorAddress"
+          :index="index"
+          :value="validator"
```
Comment (nwingt) [inline components/Modal/Restake.vue]: Not sure value can be assigned object

Comment (nwingt) [reply components/Modal/Restake.vue]: https://vuejs.org/v2/guide/forms.html#Select-Options

Seems ok

File: components/Modal/Restake.vue [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+        >
+          {{ enhancedDestinationValidator(validator) }}
+        </option>
+      </select>
```
Comment (nwingt) [inline components/Modal/Restake.vue]: I think can use `CommonField` with `options` props
https://github.com/likecoin/lunie-ng/blob/43c56c518fe7a8209c4e923cceef7e9ade4e7276/components/Modal/Send.vue#L61-L67

File: pages/validators/_address.vue [hunk 3]
```diff
@@ -313,10 +328,14 @@ h5 {
...
 
-.action-buttons button:first-child {
+.action-buttons button {
   margin-right: 0.5rem;
 }
 
+.action-buttons button:last-child {
+  margin-right: 0;
+}
+
```
Comment (nwingt) [inline pages/validators/_address.vue]:
```suggestion
.action-buttons button:not(:last-child) {
  margin-right: 0.5rem;
}
```

File: pages/validators/_address.vue [hunk 4]
```diff
@@ -38,6 +38,12 @@
             type="secondary"
             @click.native="openUnstakeModal"
           />
+          <CommonButton
+            :disabled="!delegation"
+            :value="`Restake`"
```
Comment (nwingt) [inline pages/validators/_address.vue:43]:
```suggestion
            value="Restake"
```

File: components/Modal/Restake.vue [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+    },
+    enterPressed() {
+      this.$refs.ModalAction.validateChangeStep()
+    },
+    onSuccess(event) {
+      this.$emit(`success`, event)
+    },
+    enhancedDestinationValidator(destinationValidator) {
+      const prefix =
+        destinationValidator.status === 'INACTIVE' ? '(inactive) ' : ''
```
Comment (nwingt) [inline components/Modal/Restake.vue]:
```suggestion
        destinationValidator.status === 'INACTIVE' ? '(Inactive) ' : ''
```

File: components/Modal/Restake.vue [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+      }
+    },
+    enhancedSourceValidator() {
+      return validatorEntry(this.sourceValidator)
+    },
+    sortedEnrichedValidators() {
+      const orderedValidators = orderBy(
+        this.validators,
+        [this.sort.property],
+        [this.sort.order]
```
Comment (nwingt) [inline components/Modal/Restake.vue]: Can remove `sort` in data
```suggestion
        [`votingPower`],
        [`desc`]
```

**Review by nwingt — CHANGES_REQUESTED** [review]

File: pages/validators/_address.vue (hunk 4 above)
Comment (WeiJun0827) [reply pages/validators/_address.vue:43]: The original author uses the same coding style like:
```
<CommonButton :value="`Stake`" @click.native="openStakeModal" />
<CommonButton
  :disabled="!delegation"
  :value="`Unstake`"
  type="secondary"
  @click.native="openUnstakeModal"
/>
```
Is it preferred to follow?

File: pages/validators/_address.vue (hunk 3 above)
Comment (WeiJun0827) [reply pages/validators/_address.vue]: Fixed

File: components/Modal/Restake.vue (hunk 5 above)
Comment (WeiJun0827) [reply components/Modal/Restake.vue]: Fixed

File: components/Modal/Restake.vue (hunk 6 above)
Comment (WeiJun0827) [reply components/Modal/Restake.vue]: Fixed

File: pages/validators/_address.vue (hunk 4 above)
Comment (nwingt) [reply pages/validators/_address.vue:43]: Then follow their style

**Review by nwingt — APPROVED** [review]

**Pull request merged**